sdk/search/ai-search/README.md

```diff
@@ -0,0 +1,302 @@
+
+```js
+const page = (
+  await client
+    .listSearchResults("WiFi", {
+      facets: ["Category,count:3,sort:count", "Rooms/BaseRate,interval:100"]
+    })
+    .byPage()
+    .next()
+).value;
+
+// {
+//   Rating: [
+//     { count: 4, value: 2 },
+//     { count: 5, value: 3 },
+//     { count: 8, value: 4 }
+//   ],
+//   'Rooms/BaseRate': [
+//     { count: 16, value: 0 },
+//     { count: 17, value: 100 },
+//     { count: 17, value: 200 }
+//   ],
+//   Category: [
+//     { count: 5, value: 'Budget' },
+//     { count: 5, value: 'Luxury' },
+//     { count: 5, value: 'Resort and Spa' }
+//   ]
+// }
+console.log(page.facets);
+```
+
+When retrieving results, an `facets` property will be available that will indicate the number of results that fall into each facet bucket. This can be used to drive refinement (e.g. issuing a follow-up search that filters on the `Rating` being greater than 3 and less than 4.)
+
+### Retrieving documents by id
+
+A specific document can be retrieved by its primary key value:
+
+```js
+const result = await client.getDocument("1234");
+console.log(result);
+```
+
+### Retrieve suggestions from an index
+
+If you [created a suggester](https://docs.microsoft.com/en-us/azure/search/index-add-suggesters) on your index, you can use it to return result suggestions for a user query.
+
+This example shows returning the top three suggestions for the input "wifi" from the suggester "sg":
+
+```js
+const suggestResult = await client.suggest("sg", "wifi", {
+  select: ["HotelId", "HotelName"],
+  highlightPreTag: "<em>",
+  highlightPostTag: "</em>",
+  top: 3
+});
+
+for (const result of suggestResult.results) {
+  console.log(`Suggestion: ${result.HotelName}; Match text: ${result.text}`);
+}
+```
+
+### Autocomplete a partial query using an index
+
+To implement type-ahead behavior in your application, you can query the index with partial user input and return a list of suggested completions.
+
+The below example tries to complete the string "de" using the suggester named "sg" on the index:
+
+```js
+const autocompleteResult = await client.autocomplete("sg", "de");
+
+for (const result of autocompleteResult.results || []) {
+  console.log(result.text);
+}
+```
+
+### Return the count of documents in an index
+
+```js
+const count = await client.count();
+console.log(`${count} documents in index ${client.indexName}`);
+```
+
+### Delete documents in an index
+
+Given the name of a primary key and a list of indexes, you can delete multiple documents from the index at the same time:
+
+```js
+const deleteResult = await client.deleteDocuments("HotelId", ["1", "2", "3"]);
+for (const result of deleteResult.results) {
+  console.log(`Deleting ${result.key}; succeeded? ${result.succeeded}`);
+}
+```
+
+### Upload documents into an index
+
+You can upload multiple documents into index inside a batch:
+
+```js
+const uploadResult = await client.uploadDocuments([
+  { ... },
```

Comment:
Is this saying the documents can be any structure? Does it just get JSONify'd?

Reply:
It has to match the schema of your index. I just didn't think it would be that interesting for me to showcase fake hotel data here.

Reply:
Might not - perhaps just a comment in there to indicate that this is their schema and document and not some wrapper. For instance, in eventhubs we have this required envelope - when I was looking at what you had there I wasn't sure if that was the case or not.

Reply:
Gotcha. Will add a clarifying comment.